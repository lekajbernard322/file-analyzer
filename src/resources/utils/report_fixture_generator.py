from os import listdir
from os.path import isfile, join
import json
from datetime import datetime, timezone
from bson.objectid import ObjectId

INPUT_DIR = r'../report_data/'
OUTPUT_PATH = r'../../backend/app/report/fixtures/report_fixture.json'

file_paths = [(join(INPUT_DIR, file_name), file_name)
              for file_name in listdir(INPUT_DIR)
              if isfile(join(INPUT_DIR, file_name))]


def prepare_output_data(i, file_name, content):
    return {
        'model': 'report.report',
        'pk': str(ObjectId()),
        'fields': {
            'name': file_name,
            'content': content,
            'created': datetime.now(tz=timezone.utc).isoformat()
        }
    }


data = []
for index, (input_file_path, input_file_name) in enumerate(file_paths):
    print(f'[*] Processing file: {input_file_path}')

    with open(input_file_path, 'r') as input_file:
        json_content = json.load(input_file)

        output_data = prepare_output_data(index, input_file_name, json_content)

        data.append(output_data)

print(f'[*] Dumping to fixture file: {OUTPUT_PATH}')
with open(OUTPUT_PATH, 'w', encoding='utf-8') as output_file:
    json.dump(data, output_file, indent=4)

print(f'[\u2714] Done')
