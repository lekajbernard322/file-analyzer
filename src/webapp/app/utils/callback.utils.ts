export const createOnPressEnter = (callback) => ({key}) => key === 'Enter' && callback && callback()
