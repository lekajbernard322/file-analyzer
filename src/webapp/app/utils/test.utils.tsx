import thunk from "redux-thunk";
import {createMemoryHistory} from "history";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import React from "react";
import configureStore from "redux-mock-store";

export const renderWithContext = (additionalState, ui) => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(additionalState);
    const history = createMemoryHistory();

    const Wrapper = ({children}) => (
        <Provider store={store}>
            <Router history={history}>
                {children}
            </Router>
        </Provider>);

    return {rendered: render(ui, {wrapper: Wrapper}), store, history};
};
