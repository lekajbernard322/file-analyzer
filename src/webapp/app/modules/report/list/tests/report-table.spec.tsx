import React from 'react';
import { fireEvent } from "@testing-library/react";
import ReportTable from "../report-table";
import {renderWithContext} from "../../../../utils/test.utils";

const onLoadMore = jest.fn();

const mockReports = [
    {
        _id: "1",
        name: "report1",
        created: "some date",
    },
    {
        _id: "2",
        name: "report1",
        created: "some date",
    },
];

describe("ReportTable", () => {

    const renderComponent = (mockReports, mockDisabled) => {
        return renderWithContext(null, <ReportTable reports={mockReports} disabled={mockDisabled} onLoadMore={onLoadMore}/>)
    }

    beforeAll(() => {
        onLoadMore.mockClear();
    })

    it('should render correctly with data', () => {
        const {rendered: {container}} = renderComponent(mockReports, false)

        expect(container).toMatchSnapshot();
    });

    it('should render no data message when no reports are available', () => {
        const {rendered: {container}} = renderComponent(null, false)

        expect(container).toMatchSnapshot();
    });

    it('should call onLoadMore when clicked on load more button', () => {
        const {rendered: {queryByText}} = renderComponent(null, false);

        const loadMoreButton = queryByText('Load more');

        fireEvent.click(loadMoreButton);

        expect(onLoadMore).toHaveBeenCalled();
    });

    it('should call onLoadMore when clicked on load more button', () => {
        const {rendered: {queryByText}} = renderComponent(null, true);

        const loadMoreButton = queryByText('Load more');

        expect(loadMoreButton).toBeDisabled();
    });
});
