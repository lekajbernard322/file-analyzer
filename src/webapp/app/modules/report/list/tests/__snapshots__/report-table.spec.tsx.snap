// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTable should render correctly with data 1`] = `
<div>
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Created
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            report1
          </td>
          <td>
            some date
          </td>
          <td
            class="actions"
          >
            <a
              href="/details/1"
            >
              View
            </a>
          </td>
        </tr>
        <tr>
          <td>
            report1
          </td>
          <td>
            some date
          </td>
          <td
            class="actions"
          >
            <a
              href="/details/2"
            >
              View
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <button>
      Load more
    </button>
  </div>
</div>
`;

exports[`ReportTable should render no data message when no reports are available 1`] = `
<div>
  <div>
    <table>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Created
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            colspan="3"
          >
            No reports found
          </td>
        </tr>
      </tbody>
    </table>
    <button>
      Load more
    </button>
  </div>
</div>
`;
